org.eclipse.jdt.text.tests/src/org/eclipse/jdt/text/tests/folding/CustomFoldingRegionNewFoldingTest.java

```diff
@@ -0,0 +1,189 @@
+/*******************************************************************************
```

Comment:
This test class runs all tests from `testCommentsInEmptyBlocks` with the new folding introduced in #1562.
The test methods that yield different results (due to the folding being different in some ways, for example #1992) are overridden.